include/open62541/plugin/accesscontrol.h

```diff
@@ -80,27 +108,36 @@ struct UA_AccessControl {
     /* Allow adding a node */
     UA_Boolean (*allowAddNode)(UA_Server *server, UA_AccessControl *ac,
                                const UA_NodeId *sessionId, void *sessionContext,
-                               const UA_AddNodesItem *item);
+                               const UA_AddNodesItem *item, UA_RolePermissionType *userRolePermission, size_t userRoleSize);
 
     /* Allow adding a reference */
     UA_Boolean (*allowAddReference)(UA_Server *server, UA_AccessControl *ac,
                                     const UA_NodeId *sessionId, void *sessionContext,
-                                    const UA_AddReferencesItem *item);
+                                    const UA_AddReferencesItem *item, UA_RolePermissionType *userRolePermission, size_t userRoleSize);
 
     /* Allow deleting a node */
     UA_Boolean (*allowDeleteNode)(UA_Server *server, UA_AccessControl *ac,
                                   const UA_NodeId *sessionId, void *sessionContext,
-                                  const UA_DeleteNodesItem *item);
+                                  const UA_DeleteNodesItem *item, UA_RolePermissionType *userRolePermission,size_t userRolePermissionSize);
 
     /* Allow deleting a reference */
     UA_Boolean (*allowDeleteReference)(UA_Server *server, UA_AccessControl *ac,
                                        const UA_NodeId *sessionId, void *sessionContext,
-                                       const UA_DeleteReferencesItem *item);
+                                       const UA_DeleteReferencesItem *item, UA_RolePermissionType *userRolePermission, size_t userRoleSize);
 
     /* Allow browsing a node */
     UA_Boolean (*allowBrowseNode)(UA_Server *server, UA_AccessControl *ac,
                                   const UA_NodeId *sessionId, void *sessionContext,
-                                  const UA_NodeId *nodeId, void *nodeContext);
+                                  const UA_NodeId *nodeId, void *nodeContext, UA_RolePermissionType *userRolePermission,size_t userRoleSize);
+
+    /* Check access to Node */
+    UA_Boolean (*hasAccessToNode)(UA_Server *server, UA_AccessControl *ac,
```

Comment:
What is the difference between `hasAccessToNode` and `getUserRightsMask`?
This seems to be a duplicate.